cpp/src/lance/io/scanner.h

```diff
@@ -39,7 +39,9 @@ class Scanner {
  public:
   /// Constructor.
   Scanner(std::shared_ptr<FileReader> reader,
-          std::shared_ptr<::arrow::dataset::ScanOptions> options) noexcept;
+          std::shared_ptr<::arrow::dataset::ScanOptions> options,
+          std::optional<int64_t> limit = std::nullopt,
+          int64_t offset = 0) noexcept;
```

Comment:
Why is limit an optional but offset just has a default 0 value?

Reply:
I was choosing to make limit optional because limit does not have a valid default value, while offset = 0 is legit value.  Don't want to make `limit = -1` for default null value for static checking purpose.